src/EditorFeatures/CSharpTest/Completion/CompletionProviders/PropertySubPatternCompletionProviderTests.cs

```diff
@@ -0,0 +1,653 @@
+        {
+            var markup =
+@"
+class Program
+{
+    public int P1 { get; set; }
+    public int P2 { get; set; }
+
+    void M()
+    {
+        _ = this is Program { P2: 1, $$ }
+    }
+}
+";
+            // VerifyItemExistsAsync also tests with the item typed.
+            await VerifyItemExistsAsync(markup, "P1");
+            await VerifyItemIsAbsentAsync(markup, "P2");
+        }
+
+        [Fact]
+        public async Task PropertiesInRecursivePattern_PositionalInFirstProperty()
+        {
+            var markup =
+@"
+class Program
+{
+    public D P1 { get; set; }
+
+    void M()
+    {
+        _ = this is Program { P1: ($$ }
+    }
+}
+class D
+{
+    public int P2 { get; set; }
+}
+";
+            await VerifyNoItemsExistAsync(markup);
+        }
+
+        [Fact]
+        public async Task PropertiesInRecursivePattern_PositionalInFirstProperty_AfterComma()
+        {
+            var markup =
+@"
+class Program
+{
+    public D P1 { get; set; }
+
+    void M()
+    {
+        _ = this is Program { P1: (1, $$ }
+    }
+}
+class D
+{
+    public int P2 { get; set; }
+}
+";
+            await VerifyNoItemsExistAsync(markup);
+        }
+
+        [Fact]
+        public async Task PropertiesInRecursivePattern_PositionalInFirstProperty_AfterCommaAndBeforeParen()
+        {
+            var markup =
+@"
+class Program
+{
+    public D P1 { get; set; }
+
+    void M()
+    {
+        _ = this is Program { P1: (1, $$) }
+    }
+}
+class D
+{
+    public int P2 { get; set; }
+}
+";
+            await VerifyNoItemsExistAsync(markup);
+        }
+
+        [Fact]
+        public async Task PropertiesInRecursivePattern_InPositional_Incomplete()
+        {
+            var markup =
+@"
+public class Program
+{
+    public int P1 { get; set; }
+
+    void M()
+    {
+        _ = this is ({ $$ }) // no deconstruction into 1 element
+    }
+
+    public void Deconstruct(out Program x, out Program y) => throw null;
+}
+";
+            // PROTOTYPE(patterns2): Like for typing arguments in methods, we should fall back to best overload resolution candidate
+            await VerifyNoItemsExistAsync(markup);
+        }
+
+        [Fact]
+        public async Task PropertiesInRecursivePattern_InPositional_Complete_BeforeComma()
+        {
+            var markup =
+@"
+public class Program
+{
+    public int P1 { get; set; }
+
+    void M()
+    {
+        _ = this is ({ $$ }, )
+    }
+
+    public void Deconstruct(out Program x, out Program y) => throw null;
+}
+";
+            await VerifyItemExistsAsync(markup, "P1");
+        }
+
+        [Fact]
+        public async Task PropertiesInRecursivePattern_InPositional_Complete_AfterComma()
+        {
+            var markup =
+@"
+public class Program
+{
+    public int P1 { get; set; }
+
+    void M()
+    {
+        _ = this is ( , { $$ })
+    }
+
+    public void Deconstruct(out Program x, out Program y) => throw null;
```

Comment:
consider changing this test or adding a new one where these two types are different